File: pc-bridge/tools/inspect_state.py
import os, sqlite3, sys

home = os.path.expanduser('~')
paths = [
    os.path.join(home, '.codex', 'state_5.sqlite'),
    os.path.join(home, '.codex', 'sqlite', 'state_5.sqlite'),
]
for path in paths:
    print('=== %s ===' % path)
    if not os.path.exists(path):
        print('missing')
        continue
    try:
        con = sqlite3.connect('file:%s?mode=ro' % path, uri=True)
        cur = con.cursor()
        cur.execute("SELECT name FROM sqlite_master WHERE type='table'")
        tables = [r[0] for r in cur.fetchall()]
        for name in tables:
            low = name.lower()
            if not any(k in low for k in ('thread', 'host', 'session')):
                continue
            print('\n-- table %s --' % name)
            cur.execute('PRAGMA table_info(%s)' % name)
            cols = [r[1] for r in cur.fetchall()]
            print('cols:', ', '.join(cols))
            try:
                cur.execute('SELECT * FROM %s LIMIT 5' % name)
                for row in cur.fetchall():
                    print(row)
            except Exception as e:
                print('select error:', e)
        con.close()
    except Exception as e:
        print('error:', e)

File: pc-bridge/tools/inspect-state.mjs
import { createRequire } from 'node:module';
import { join } from 'node:path';
import { homedir } from 'node:os';

const require = createRequire(import.meta.url);
const Database = require(join(process.env.TEMP || 'C:/Users/0000/AppData/Local/Temp', 'codex-asar-extract', 'node_modules', 'better-sqlite3'));

const candidates = [
  join(homedir(), '.codex', 'state_5.sqlite'),
  join(homedir(), '.codex', 'sqlite', 'state_5.sqlite'),
];

for (const path of candidates) {
  console.log('=== ' + path + ' ===');
  try {
    const db = new Database(path, { readonly: true });
    const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table'").all();
    for (const { name } of tables) {
      if (!/thread|host|session/i.test(name)) continue;
      console.log('\n-- table ' + name + ' --');
      const cols = db.prepare(`PRAGMA table_info(${name})`).all().map((c) => c.name);
      console.log('cols:', cols.join(', '));
      const rows = db.prepare(`SELECT * FROM ${name} LIMIT 5`).all();
      for (const row of rows) console.log(JSON.stringify(row));
    }
    db.close();
  } catch (error) {
    console.log('error:', error.message);
  }
}